### DISCUSSION TITLE
Adding custom models to TTS engines (F5 TTS)

### DISCUSSION THREAD

**Discussion by M4x-P0wer** [erew123/alltalk_tts#471] [Q&A] [answered]
Hello alltalk Community,

I am wondering if it is possible to add a custom model to the available TTS engines.
As an example, I have a HF F5-TTS model (https://huggingface.co/marduk-ra/F5-TTS-German) that I would like to use via alltalk.

Is this possible?
I am unable to select the model if I try to add the related files to the directory.

**Comment by erew123** [comment]
Hi @M4x-P0wer 

Yes you should be able to add any model to any of the TTS engines. I go by what the developer of the TTS engine says you need for file names etc:

https://github.com/erew123/alltalk_tts/blob/alltalkbeta/system/tts_engines/f5tts/model_engine.py#L402-L410

So I dont know which one of those models you are wanting to use, but you would need to rename them from `f5_tts_german_1010000.safetensors` to `model_f5_tts_german_1010000.safetensors` (basically add the word `model` to the start as AllTalk is looking for a model file with the word `model` at the start)

Make sure you also have the vocab.txt in the same folder. 

Finally you will need the `vocos` folder too with its 2x files in. Im assuming that using the default ones from the default model will be fine.


![image](https://github.com/user-attachments/assets/6359baf5-04c4-4aaa-9b83-1bf16c1b699d)
![image](https://github.com/user-attachments/assets/90afb756-94d6-4888-a453-28bfacf2a453)

**Reply by M4x-P0wer** [reply]
Thank you @erew123 and a happy new year!
Renaming the model (model_*) and directory (tts_v *) made it work.

**Comment by M4x-P0wer** [comment, marked as answer]
Renaming the model (model_*) and directory (*_tts_v *) made it work.
